src/main/java/io/vertx/openapi/contract/OpenAPIVersion.java

```diff
@@ -75,16 +76,42 @@ public static OpenAPIVersion fromContract(JsonObject contract) {
     }
   }
 
-  public Future<OutputUnit> validate(Vertx vertx, SchemaRepository repo, JsonObject contract) {
-    return vertx.executeBlocking(p -> p.complete(repo.validator(mainSchemaFile).validate(contract)));
+  public Future<OutputUnit> validateContract(Vertx vertx, SchemaRepository repo, JsonObject contract) {
+    return vertx.executeBlocking(() -> repo.validator(mainSchemaFile).validate(contract));
+  }
+
+  /**
+   * Validates additional contract files against the openapi schema. If validations fails, try to validate against the
+   * json schema specifications only.
+   *
+   * @param vertx                   The related Vert.x instance.
+   * @param repo                    The SchemaRepository to do the validations with.
+   * @param file                    The additional json contract to validate.
+   */
+  public Future<Void> validateAdditionalContractFiles(Vertx vertx, SchemaRepository repo, JsonObject file) {
```

Comment:
In my opinion `validateAdditionalContractFiles` is missleading, as it validates only a single file. we should name it `validateAdditionalContractFile`. Or pass all files.

This method should have its own unit test in `OpenAPIVersionTest`. The `testValidate` method in `OpenAPIVersionTest` should get renamed in `validateContractTest` and the new test method should be called `validateAdditionalContractFileTest`.

Reply:
Updated these and added an additional test for validateAdditionalContractFile itself.